x/maxsupply/keeper/keeper.go

```diff
@@ -0,0 +1,117 @@
+package keeper
+
+import (
+	"context"
+	"errors"
+
+	"github.com/crypto-org-chain/chain-main/v4/x/maxsupply/types"
+
+	"cosmossdk.io/core/store"
+	"cosmossdk.io/log"
+	"cosmossdk.io/math"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type Keeper struct {
+	cdc           codec.BinaryCodec
+	storeService  store.KVStoreService
+	logger        log.Logger
+	bankKeeper    types.BankKeeper
+	stakingKeeper types.StakingKeeper
+
+	// the address capable of executing a MsgUpdateParams message. Typically, this
+	// should be the x/gov module account.
+	authority string
+}
+
+// NewKeeper creates a new maxsupply Keeper instance
+func NewKeeper(
+	cdc codec.BinaryCodec,
+	storeService store.KVStoreService,
+	logger log.Logger,
+	bankKeeper types.BankKeeper,
+	stakingKeeper types.StakingKeeper,
+	authority string,
+) Keeper {
+	return Keeper{
+		cdc:           cdc,
+		storeService:  storeService,
+		logger:        logger,
+		bankKeeper:    bankKeeper,
+		stakingKeeper: stakingKeeper,
+		authority:     authority,
+	}
+}
+
+// Logger returns a module-specific logger.
+func (k Keeper) Logger(ctx context.Context) log.Logger {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	return k.logger.With("module", "x/"+types.ModuleName, "height", sdkCtx.BlockHeight())
+}
+
+// GetParams get all parameters as types.Params
+func (k Keeper) GetParams(ctx context.Context) (params types.Params) {
+	store := k.storeService.OpenKVStore(ctx)
+	bz, err := store.Get([]byte(types.ParamsKey))
+	if err != nil {
+		panic(err)
```

Comment:
Alternatively, return an error here and convert to panic only at the `beginblock`

May be safer if we reuse this method somewhere else in the future